storage/clickhouse/clickhouse.go

```diff
@@ -277,6 +281,54 @@ func (ch *clickHouse) convertReadRequest(request *prompb.Query) base.Query {
 	return q
 }
 
+func (ch *clickHouse) prepareClickHouseQuery(query *prompb.Query, metricName string) (string, error) {
+	var clickHouseQuery string
+	var conditions []string
+	conditions = append(conditions, fmt.Sprintf("metric_name = '%s'", metricName))
+	for _, m := range query.Matchers {
+		switch m.Type {
+		case prompb.LabelMatcher_EQ:
+			conditions = append(conditions, fmt.Sprintf("JSONExtractString(labels, '%s') = '%s'", m.Name, m.Value))
+		case prompb.LabelMatcher_NEQ:
+			conditions = append(conditions, fmt.Sprintf("JSONExtractString(labels, '%s') != '%s'", m.Name, m.Value))
+		case prompb.LabelMatcher_RE:
+			conditions = append(conditions, fmt.Sprintf("match(JSONExtractString(labels, '%s'), '%s')", m.Name, m.Value))
+		case prompb.LabelMatcher_NRE:
+			conditions = append(conditions, fmt.Sprintf("not match(JSONExtractString(labels, '%s'), '%s')", m.Name, m.Value))
+		default:
+			return "", fmt.Errorf("prepareClickHouseQuery: unexpected matcher %d", m.Type)
+		}
+	}
+	whereClause := strings.Join(conditions, " AND ")
+
+	clickHouseQuery = fmt.Sprintf(`SELECT fingerprint FROM %s.time_series_v2 WHERE %s`, ch.database, whereClause)
```

Comment:
Should this be `SELECT DISTINCT fingerprint ...`?

Reply:
use prepared statements or arg based query rather than constructing a string for db query. This will help prevent sql injection

Reply:
I will tackle this prepared statements work across the repos in another issue.

Reply:
ok